File: 729-my-calendar-i/729-my-calendar-i.cpp
class MyCalendar {
    List<int[]> cal ; 
    
    
    public MyCalendar() {
        cal = new ArrayList() ; 
        
    }
    
    public boolean book(int s, int e) {
        
        for(int[] se: cal){
            if(se[0] < e && s < se[1] ) return false ; 
        }
        cal.add(new int[]{s ,e }) ;
        return true ; 
        
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * boolean param_1 = obj.book(start,end);
 */

File: 729-my-calendar-i/729-my-calendar-i.py
class MyCalendar:

    def __init__(self):
        self.record = [] 

    def book(self, ns: int, ne: int) -> bool:
        
        for cs, ce in self.record: 
            
            if (ns < cs and cs < ne ) or  ( ce < ne and ce > ns ) or (cs <=ns and ne <=ce): 
                return False 
        
        self.record.append([ns,ne])
        return True 


# Your MyCalendar object will be instantiated and called as such:
# obj = MyCalendar()
# param_1 = obj.book(start,end)